godtools/App/Features/ToolsFilter/Presentation/ToolFilterLanguageSelectionView/ToolFilterLanguageSelectionViewModel.swift

```diff
@@ -43,22 +45,37 @@ class ToolFilterLanguageSelectionViewModel: ObservableObject {
         self.selectedLanguage = languageFilterSelectionPublisher.value
         self.flowDelegate = flowDelegate
         
+        getCurrentAppLanguageUseCase
+            .getLanguagePublisher()
+            .assign(to: &$appLanguage)
+        
+        $appLanguage.eraseToAnyPublisher()
+            .flatMap { appLanguage in
+                
+                return viewToolFilterLanguagesUseCase.viewPublisher(filteredByCategoryId: selectedCategory.id, translatedInAppLanguage: appLanguage)
+            }
+            .receive(on: DispatchQueue.main)
+            .sink { viewLanguageFiltersDomainModel in
+                
+                self.allLanguages = viewLanguageFiltersDomainModel.languageFilters
+            }
+            .store(in: &cancellables)
+
         getInterfaceStringInAppLanguageUseCase
             .getStringPublisher(id: ToolStringKeys.ToolFilter.languageFilterNavTitle.rawValue)
             .receive(on: DispatchQueue.main)
             .assign(to: &$navTitle)
         
-        getToolFilterLanguagesUseCase.getToolFilterLanguagesPublisher(filteredByCategory: selectedCategory)
-            .receive(on: DispatchQueue.main)
-            .assign(to: &$allLanguages)
-        
-        searchToolFilterLanguagesUseCase
-            .getSearchResultsPublisher(
-                for: $searchText.eraseToAnyPublisher(),
-                in: $allLanguages.eraseToAnyPublisher()
-            )
-            .receive(on: DispatchQueue.main)
-            .assign(to: &$languageSearchResults)
+        Publishers.CombineLatest(
+            $searchText.eraseToAnyPublisher(),
+            $allLanguages.eraseToAnyPublisher()
+        )
+        .flatMap { searchText, allLanguages in
+            
+            return self.searchToolFilterLanguagesUseCase.getSearchResultsPublisher(for: searchText, in: allLanguages)
```

Comment:
Capturing self here in flatMap will cause a retain cycle.  Easily resolved by using searchToolFilterLanguagesUseCase that is in the scope of init.